cmd/srcd/cmd/sql_test.go

```diff
@@ -5,24 +5,225 @@ package cmd
+| repositories |
+| tree_entries |
++--------------+
+`)
+
+var showRepoTableDescOutput = sqlOutput(`+---------------+------+
+| name          | type |
++---------------+------+
+| repository_id | TEXT |
++---------------+------+
+`)
+
+type SQLREPLTestSuite struct {
+	cmdtest.IntegrationSuite
+	testDir string
+}
+
+func TestSQLREPLTestSuite(t *testing.T) {
+	s := SQLREPLTestSuite{}
+	suite.Run(t, &s)
+}
+
+func (s *SQLREPLTestSuite) SetupTest() {
+	var err error
+	s.testDir, err = ioutil.TempDir("", "sql-repl-test")
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func (s *SQLREPLTestSuite) TearDownTest() {
+	s.RunCommand(context.Background(), "prune")
+	os.RemoveAll(s.testDir)
+}
+
+func (s *SQLREPLTestSuite) TestREPL() {
+	// When it is not in a terminal, the command reads stdin and exits.
+	// You can see it running:
+	//     $ echo "show tables;" | ./srcd sql
+	// So this test does not really interacts with the REPL prompt, but we still
+	// test that the code that processes each read line is working as expected.
+
+	require := s.Require()
+
+	var out, in bytes.Buffer
+
+	command := s.CommandContext(context.TODO(), "sql")
+	command.Stdout = &out
+	command.Stderr = &out
+	command.Stdin = &in
+
+	io.WriteString(&in, "show tables;\n")
+	io.WriteString(&in, "describe table repositories;\n")
+
+	err := command.Start()
+	require.NoError(err)
+
+	err = command.Wait()
+	require.NoError(err)
+
+	require.Contains(out.String(), showTablesOutput)
+	require.Contains(out.String(), showRepoTableDescOutput)
+}
+
+func (s *SQLREPLTestSuite) TestInteractiveREPL() {
+	require := s.Require()
+
+	command, stdin, ch := s.runInteractiveRepl()
+
+	res := s.runInteractiveQuery(stdin, "show tables;\n", ch)
+	require.Contains(res, showTablesOutput)
+
+	res = s.runInteractiveQuery(stdin, "describe table repositories;\n", ch)
+	require.Contains(res, showRepoTableDescOutput)
+
+	require.NoError(s.exitInteractiveAndWait(10*time.Second, stdin, ch))
+	require.NoError(s.waitMysqlCliContainerStopped(10 * time.Second))
+
+	command.Wait()
+}
+
+func (s *SQLREPLTestSuite) runInteractiveRepl() (*exec.Cmd, *os.File, <-chan string) {
+	s.T().Helper()
+
+	command := s.CommandContext(context.TODO(), "sql")
+
+	ch := make(chan string)
+	cr := cmdtest.NewChannelWriter(ch)
+
+	command.Stdout = cr
+	command.Stderr = cr
+
+	f, err := pty.Start(command)
+	if err != nil {
+		panic(err)
+	}
+
+	mysqlCliReady := make(chan struct{})
+
+	go func() {
+		for s := range ch {
+			lines := strings.Split(s, "\n")
+			for _, l := range lines {
+				if strings.HasPrefix(l, "mysql>") {
+					mysqlCliReady <- struct{}{}
+					return
+				}
+			}
+		}
+	}()
+
+	<-mysqlCliReady
+
+	return command, f, ch
+}
+
+func (s *SQLREPLTestSuite) runInteractiveQuery(stdin *os.File, query string, ch <-chan string) string {
+	io.WriteString(stdin, query)
+
+	var res strings.Builder
+	for c := range ch {
+		fmt.Printf("Message: %s\n", c)
+		res.WriteString(c)
+		if s.containsSQLOutput(res.String()) {
+			break
```

Comment:
looks like you expect a full output of the command in `c`. Which isn't true right now.
It can contain only
```
+--------------+
| Table        |
+--------------+
| blobs        |
```
for example